akka-remote/src/main/scala/akka/remote/artery/AeronSink.scala

```diff
@@ -105,7 +108,17 @@ class AeronSink(channel: String, streamId: Int, aeron: Aeron, taskRunner: TaskRu
       }
 
       private def onOfferSuccess(): Unit = {
-        pull(in)
+        offerTaskInProgress = false
+        if (isClosed(in))
+          completeStage()
+        else
+          pull(in)
+      }
+
+      override def onUpstreamFinish(): Unit = {
```

Comment:
I don't think this will work, I think you need to set `setKeepGoing(true)`, because the default behavior is that if all ports of a stage has been closed then it stops itself.